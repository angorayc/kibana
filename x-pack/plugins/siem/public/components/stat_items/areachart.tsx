/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { pure } from 'recompose';
import styled from 'styled-components';
// @ts-ignore
import { EuiSeriesChart, EuiAreaSeries, EuiXAxis, EuiYAxis } from '@elastic/eui/lib/experimental';
import { clone as _clone } from 'lodash';
import { AreaChartData, WrappedByAutoSizer, ChartHolder } from '.';
import { AutoSizer } from '../auto_sizer';

const ChartBaseComponent = pure<{
  data: AreaChartData[];
  width: number | undefined;
  height: number | undefined;
}>(({ data, ...chartConfigs }) => {
  return chartConfigs.width &&
    chartConfigs.height &&
    data &&
    data.length &&
    data.every(({ value }) => value != null && value.length > 0) ? (
    <SeriesChart {...chartConfigs} showDefaultAxis={false} xType="ordinal">
      {data.map(series => {
        return (
          // @ts-ignore
          <EuiAreaSeries
            key={`stat-items-areachart-${series.key}`}
            name={series.key.replace('Histogram', '')}
            data={series.value!}
            fillOpacity={0.04}
            color={series.color}
          />
        );
      })}
      {/*
// @ts-ignore */}
      <EuiXAxis tickFormat={value => value.toString().split('T')[0]} />
      {/*
// @ts-ignore */}
      <EuiYAxis />
    </SeriesChart>
  ) : (
    <ChartHolder />
  );
});

export const AreaChart = pure<{ areaChart: AreaChartData[] }>(({ areaChart }) => (
  <AutoSizer detectAnyWindowResize={false} content>
    {({ measureRef, content: { height, width } }) => (
      <WrappedByAutoSizer data-test-subj="wrapped-by-auto-sizer" innerRef={measureRef}>
        <ChartBaseComponent height={height} width={width} data={areaChart} />
      </WrappedByAutoSizer>
    )}
  </AutoSizer>
));

// @ts-ignore
const SeriesChart = styled(EuiSeriesChart)`
  svg .rv-xy-plot__axis__ticks .rv-xy-plot__axis__tick:not(:first-child):not(:last-child) {
    display: none;
  }
`;
