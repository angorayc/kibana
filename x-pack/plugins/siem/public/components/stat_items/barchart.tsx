/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import {
  // @ts-ignore
  EuiSeriesChartUtils,
} from '@elastic/eui';
import { pure } from 'recompose';
import styled from 'styled-components';
// @ts-ignore
import { EuiSeriesChart, EuiBarSeries, EuiXAxis, EuiYAxis } from '@elastic/eui/lib/experimental';
import { BarChartData, WrappedByAutoSizer, ChartHolder } from '.';
import { AutoSizer } from '../auto_sizer';

const { SCALE, ORIENTATION } = EuiSeriesChartUtils;
const getYaxis = (value: string | number) => {
  const label = value.toString();
  return label.length > 4 ? `${label.slice(0, 4)}.` : label;
};

const ChartBaseComponent = pure<{
  data: BarChartData[];
  width: number | undefined;
  height: number | undefined;
}>(({ data, ...chartConfigs }) => {
  return chartConfigs.width &&
    chartConfigs.height &&
    data &&
    data.length &&
    data.every(({ value }) => value != null && value.length > 0) ? (
    <SeriesChart
      yType={SCALE.ORDINAL}
      orientation={ORIENTATION.HORIZONTAL}
      showDefaultAxis={false}
      {...chartConfigs}
    >
      {data.map(series => {
        return (
          // @ts-ignore
          <EuiBarSeries
            key={`stat-items-areachart-${series.key}`}
            name={series.key}
            data={series.value!}
            color={series.color}
          />
        );
      })}
      {/*
// @ts-ignore */}
      <EuiXAxis />
      {/*
// @ts-ignore */}
      <EuiYAxis tickFormat={getYaxis} />
    </SeriesChart>
  ) : (
    <ChartHolder />
  );
});

export const BarChart = pure<{ barChart: BarChartData[] }>(({ barChart }) => (
  <AutoSizer detectAnyWindowResize={false} content>
    {({ measureRef, content: { height, width } }) => (
      <WrappedByAutoSizer data-test-subj="wrapped-by-auto-sizer" innerRef={measureRef}>
        <ChartBaseComponent height={height} width={width} data={barChart} />
      </WrappedByAutoSizer>
    )}
  </AutoSizer>
));

// @ts-ignore
const SeriesChart = styled(EuiSeriesChart)`
  svg
    .rv-xy-plot__axis--horizontal
    .rv-xy-plot__axis__ticks
    .rv-xy-plot__axis__tick:not(:first-child):not(:last-child) {
    display: none;
  }
`;
