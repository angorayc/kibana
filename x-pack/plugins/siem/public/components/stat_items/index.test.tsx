/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { mount, shallow, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';

import { StatItemsComponent, StatItemsProps } from '.';
import { BarChart } from './barchart';
import { AreaChart } from './areachart';
import { EuiHorizontalRule, EuiIcon } from '@elastic/eui';
import { EuiFlexGroup } from '@elastic/eui';

describe('Stat Items', () => {
  describe('loading', () => {
    test('it renders loading icons', () => {
      const mockStatItemsData: StatItemsProps = {
        fields: [
          {
            key: 'networkEvents',
            description: 'NETWORK_EVENTS',
            value: null,
            color: '#000000',
          },
        ],
        isLoading: true,
        key: 'mock-key',
      };
      const wrapper = shallow(<StatItemsComponent {...mockStatItemsData} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe.each([
    [
      mount(
        <StatItemsComponent
          fields={[{ key: 'hosts', value: null, color: '#3185FC' }]}
          description="HOSTS"
          isLoading={false}
          key="mock-keys"
        />
      ),
    ],
    [
      mount(
        <StatItemsComponent
          fields={[{ key: 'hosts', value: null, color: '#3185FC' }]}
          description="HOSTS"
          isLoading={false}
          areaChart={[]}
          barChart={[]}
          key="mock-keys"
        />
      ),
    ],
  ])('rendering kpis without charts', wrapper => {
    test('it renders the default widget', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should handle multiple titles', () => {
      expect(wrapper.find('[data-test-subj="stat-title"]').filter(EuiFlexGroup)).toHaveLength(1);
    });

    test('should not render color indicators', () => {
      expect(wrapper.find(EuiIcon)).toHaveLength(0);
    });

    test('should not render barChart', () => {
      expect(wrapper.find(BarChart)).toHaveLength(0);
    });

    test('should not render areaChart', () => {
      expect(wrapper.find(AreaChart)).toHaveLength(0);
    });

    test('should not render spliter', () => {
      expect(wrapper.find(EuiHorizontalRule)).toHaveLength(0);
    });
  });

  describe('rendering kpis with charts', () => {
    const mockStatItemsData: StatItemsProps = {
      fields: [
        { key: 'uniqueSourceIps', description: 'Source', value: 1714, color: '#DB1374' },
        { key: 'uniqueDestinationIps', description: 'Dest.', value: 2359, color: '#490092' },
      ],
      areaChart: [
        {
          key: 'uniqueSourceIpsHistogram',
          value: [
            { x: 1556686800000, y: 580213 },
            { x: 1556730000000, y: 1096175 },
            { x: 1556773200000, y: 12382 },
          ],
          color: '#DB1374',
        },
        {
          key: 'uniqueDestinationIpsHistogram',
          value: [
            { x: 1556686800000, y: 565975 },
            { x: 1556730000000, y: 1084366 },
            { x: 1556773200000, y: 12280 },
          ],
          color: '#490092',
        },
      ],
      barChart: [
        { key: 'uniqueSourceIps', value: [{ x: 1714, y: 'uniqueSourceIps' }], color: '#DB1374' },
        {
          key: 'uniqueDestinationIps',
          value: [{ x: 2354, y: 'uniqueDestinationIps' }],
          color: '#490092',
        },
      ],
      description: 'UNIQUE_PRIVATE_IPS',
      isLoading: false,
      key: 'mock-keys',
    };
    let wrapper: ReactWrapper;
    beforeAll(() => {
      wrapper = mount(<StatItemsComponent {...mockStatItemsData} />);
    });
    test('it renders the default widget', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should handle multiple titles', () => {
      expect(wrapper.find('[data-test-subj="stat-title"]').filter(EuiFlexGroup)).toHaveLength(2);
    });

    test('should render color indicators', () => {
      expect(wrapper.find(EuiIcon)).toHaveLength(2);
    });

    test('should render barChart', () => {
      expect(wrapper.find(BarChart)).toHaveLength(1);
    });

    test('should render areaChart', () => {
      expect(wrapper.find(AreaChart)).toHaveLength(1);
    });

    test('should render spliter', () => {
      expect(wrapper.find(EuiHorizontalRule)).toHaveLength(1);
    });
  });

  describe('areaChart data not available', () => {
    const mockStatItemsData: StatItemsProps = {
      fields: [
        { key: 'uniqueSourceIps', description: 'Source', value: 1714, color: '#DB1374' },
        { key: 'uniqueDestinationIps', description: 'Dest.', value: 2359, color: '#490092' },
      ],
      areaChart: [
        {
          key: 'uniqueSourceIpsHistogram',
          value: [],
          color: '#DB1374',
        },
        {
          key: 'uniqueDestinationIpsHistogram',
          value: [],
          color: '#490092',
        },
      ],
      barChart: [
        { key: 'uniqueSourceIps', value: [{ x: 1714, y: 'uniqueSourceIps' }], color: '#DB1374' },
        {
          key: 'uniqueDestinationIps',
          value: [{ x: 2354, y: 'uniqueDestinationIps' }],
          color: '#490092',
        },
      ],
      description: 'UNIQUE_PRIVATE_IPS',
      isLoading: false,
      key: 'mock-keys',
    };
    let wrapper: ReactWrapper;
    beforeAll(() => {
      wrapper = mount(<StatItemsComponent {...mockStatItemsData} />);
    });

    test('should render barChart', () => {
      expect(wrapper.find(BarChart)).toHaveLength(1);
    });

    test('should not render areaChart', () => {
      expect(wrapper.find(AreaChart)).toHaveLength(0);
    });

    test('should render spliter', () => {
      expect(wrapper.find(EuiHorizontalRule)).toHaveLength(1);
    });
  });
});
