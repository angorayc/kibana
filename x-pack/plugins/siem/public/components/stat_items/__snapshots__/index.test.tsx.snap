// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Items loading it renders loading icons 1`] = `
<Component
  fields={
    Array [
      Object {
        "color": "#000000",
        "description": "NETWORK_EVENTS",
        "key": "networkEvents",
        "value": null,
      },
    ]
  }
  isLoading={true}
/>
`;

exports[`Stat Items rendering kpis with charts it renders the default widget 1`] = `
<pure(Component)
  areaChart={
    Array [
      Object {
        "color": "#DB1374",
        "key": "uniqueSourceIpsHistogram",
        "value": Array [
          Object {
            "x": 1556686800000,
            "y": 580213,
          },
          Object {
            "x": 1556730000000,
            "y": 1096175,
          },
          Object {
            "x": 1556773200000,
            "y": 12382,
          },
        ],
      },
      Object {
        "color": "#490092",
        "key": "uniqueDestinationIpsHistogram",
        "value": Array [
          Object {
            "x": 1556686800000,
            "y": 565975,
          },
          Object {
            "x": 1556730000000,
            "y": 1084366,
          },
          Object {
            "x": 1556773200000,
            "y": 12280,
          },
        ],
      },
    ]
  }
  barChart={
    Array [
      Object {
        "color": "#DB1374",
        "key": "uniqueSourceIps",
        "value": Array [
          Object {
            "x": 1714,
            "y": "uniqueSourceIps",
          },
        ],
      },
      Object {
        "color": "#490092",
        "key": "uniqueDestinationIps",
        "value": Array [
          Object {
            "x": 2354,
            "y": "uniqueDestinationIps",
          },
        ],
      },
    ]
  }
  description="UNIQUE_PRIVATE_IPS"
  fields={
    Array [
      Object {
        "color": "#DB1374",
        "description": "Source",
        "key": "uniqueSourceIps",
        "value": 1714,
      },
      Object {
        "color": "#490092",
        "description": "Dest.",
        "key": "uniqueDestinationIps",
        "value": 2359,
      },
    ]
  }
  isLoading={false}
>
  <Component
    areaChart={
      Array [
        Object {
          "color": "#DB1374",
          "key": "uniqueSourceIpsHistogram",
          "value": Array [
            Object {
              "x": 1556686800000,
              "y": 580213,
            },
            Object {
              "x": 1556730000000,
              "y": 1096175,
            },
            Object {
              "x": 1556773200000,
              "y": 12382,
            },
          ],
        },
        Object {
          "color": "#490092",
          "key": "uniqueDestinationIpsHistogram",
          "value": Array [
            Object {
              "x": 1556686800000,
              "y": 565975,
            },
            Object {
              "x": 1556730000000,
              "y": 1084366,
            },
            Object {
              "x": 1556773200000,
              "y": 12280,
            },
          ],
        },
      ]
    }
    barChart={
      Array [
        Object {
          "color": "#DB1374",
          "key": "uniqueSourceIps",
          "value": Array [
            Object {
              "x": 1714,
              "y": "uniqueSourceIps",
            },
          ],
        },
        Object {
          "color": "#490092",
          "key": "uniqueDestinationIps",
          "value": Array [
            Object {
              "x": 2354,
              "y": "uniqueDestinationIps",
            },
          ],
        },
      ]
    }
    description="UNIQUE_PRIVATE_IPS"
    fields={
      Array [
        Object {
          "color": "#DB1374",
          "description": "Source",
          "key": "uniqueSourceIps",
          "value": 1714,
        },
        Object {
          "color": "#490092",
          "description": "Dest.",
          "key": "uniqueDestinationIps",
          "value": 2359,
        },
      ]
    }
    isLoading={false}
  >
    <Styled(EuiFlexItem)
      key="stat-items-undefined"
    >
      <EuiFlexItem
        className="sc-bxivhb bQCHAr"
      >
        <div
          className="euiFlexItem sc-bxivhb bQCHAr"
        >
          <EuiPanel
            grow={true}
            hasShadow={false}
            paddingSize="m"
          >
            <div
              className="euiPanel euiPanel--paddingMedium"
            >
              <EuiTitle
                size="xxxs"
              >
                <h6
                  className="euiTitle euiTitle--xxxsmall"
                >
                  UNIQUE_PRIVATE_IPS
                </h6>
              </EuiTitle>
              <EuiFlexGroup>
                <div
                  className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <Styled(EuiFlexItem)
                    key="stat-items-field-uniqueSourceIps"
                  >
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      >
                        <EuiFlexGroup
                          alignItems="center"
                          gutterSize="m"
                          responsive={false}
                        >
                          <div
                            className="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                          >
                            <Styled(EuiFlexItem)>
                              <EuiFlexItem
                                className="sc-bxivhb bQCHAr"
                              >
                                <div
                                  className="euiFlexItem sc-bxivhb bQCHAr"
                                >
                                  <Styled(EuiTitle)>
                                    <EuiTitle
                                      className="sc-ifAKCX bfXlPM"
                                    >
                                      <p
                                        className="euiTitle euiTitle--medium sc-ifAKCX bfXlPM"
                                      >
                                        1,714
                                         
                                        Source
                                      </p>
                                    </EuiTitle>
                                  </Styled(EuiTitle)>
                                </div>
                              </EuiFlexItem>
                            </Styled(EuiFlexItem)>
                          </div>
                        </EuiFlexGroup>
                      </div>
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                  <Styled(EuiFlexItem)
                    key="stat-items-field-uniqueDestinationIps"
                  >
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      >
                        <EuiFlexGroup
                          alignItems="center"
                          gutterSize="m"
                          responsive={false}
                        >
                          <div
                            className="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                          >
                            <Styled(EuiFlexItem)>
                              <EuiFlexItem
                                className="sc-bxivhb bQCHAr"
                              >
                                <div
                                  className="euiFlexItem sc-bxivhb bQCHAr"
                                >
                                  <Styled(EuiTitle)>
                                    <EuiTitle
                                      className="sc-ifAKCX bfXlPM"
                                    >
                                      <p
                                        className="euiTitle euiTitle--medium sc-ifAKCX bfXlPM"
                                      >
                                        2,359
                                         
                                        Dest.
                                      </p>
                                    </EuiTitle>
                                  </Styled(EuiTitle)>
                                </div>
                              </EuiFlexItem>
                            </Styled(EuiFlexItem)>
                          </div>
                        </EuiFlexGroup>
                      </div>
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                </div>
              </EuiFlexGroup>
              <EuiHorizontalRule>
                <hr
                  className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
                />
              </EuiHorizontalRule>
              <EuiFlexGroup>
                <div
                  className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <Styled(EuiFlexItem)>
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      >
                        <pure(Component)
                          barChart={
                            Array [
                              Object {
                                "color": "#DB1374",
                                "key": "uniqueSourceIps",
                                "value": Array [
                                  Object {
                                    "x": 1714,
                                    "y": "uniqueSourceIps",
                                  },
                                ],
                              },
                              Object {
                                "color": "#490092",
                                "key": "uniqueDestinationIps",
                                "value": Array [
                                  Object {
                                    "x": 2354,
                                    "y": "uniqueDestinationIps",
                                  },
                                ],
                              },
                            ]
                          }
                        >
                          <Component
                            barChart={
                              Array [
                                Object {
                                  "color": "#DB1374",
                                  "key": "uniqueSourceIps",
                                  "value": Array [
                                    Object {
                                      "x": 1714,
                                      "y": "uniqueSourceIps",
                                    },
                                  ],
                                },
                                Object {
                                  "color": "#490092",
                                  "key": "uniqueDestinationIps",
                                  "value": Array [
                                    Object {
                                      "x": 2354,
                                      "y": "uniqueDestinationIps",
                                    },
                                  ],
                                },
                              ]
                            }
                          >
                            <AutoSizer
                              content={true}
                              detectAnyWindowResize={false}
                            >
                              <styled.div
                                data-test-subj="wrapped-by-auto-sizer"
                                innerRef={[Function]}
                              >
                                <div
                                  className="sc-htpNat loNVyu"
                                  data-test-subj="wrapped-by-auto-sizer"
                                >
                                  <pure(Component)
                                    data={
                                      Array [
                                        Object {
                                          "color": "#DB1374",
                                          "key": "uniqueSourceIps",
                                          "value": Array [
                                            Object {
                                              "x": 1714,
                                              "y": "uniqueSourceIps",
                                            },
                                          ],
                                        },
                                        Object {
                                          "color": "#490092",
                                          "key": "uniqueDestinationIps",
                                          "value": Array [
                                            Object {
                                              "x": 2354,
                                              "y": "uniqueDestinationIps",
                                            },
                                          ],
                                        },
                                      ]
                                    }
                                  >
                                    <Component
                                      data={
                                        Array [
                                          Object {
                                            "color": "#DB1374",
                                            "key": "uniqueSourceIps",
                                            "value": Array [
                                              Object {
                                                "x": 1714,
                                                "y": "uniqueSourceIps",
                                              },
                                            ],
                                          },
                                          Object {
                                            "color": "#490092",
                                            "key": "uniqueDestinationIps",
                                            "value": Array [
                                              Object {
                                                "x": 2354,
                                                "y": "uniqueDestinationIps",
                                              },
                                            ],
                                          },
                                        ]
                                      }
                                    />
                                  </pure(Component)>
                                </div>
                              </styled.div>
                            </AutoSizer>
                          </Component>
                        </pure(Component)>
                      </div>
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                  <Styled(EuiFlexItem)>
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      >
                        <pure(Component)
                          areaChart={
                            Array [
                              Object {
                                "color": "#DB1374",
                                "key": "uniqueSourceIpsHistogram",
                                "value": Array [
                                  Object {
                                    "x": 1556686800000,
                                    "y": 580213,
                                  },
                                  Object {
                                    "x": 1556730000000,
                                    "y": 1096175,
                                  },
                                  Object {
                                    "x": 1556773200000,
                                    "y": 12382,
                                  },
                                ],
                              },
                              Object {
                                "color": "#490092",
                                "key": "uniqueDestinationIpsHistogram",
                                "value": Array [
                                  Object {
                                    "x": 1556686800000,
                                    "y": 565975,
                                  },
                                  Object {
                                    "x": 1556730000000,
                                    "y": 1084366,
                                  },
                                  Object {
                                    "x": 1556773200000,
                                    "y": 12280,
                                  },
                                ],
                              },
                            ]
                          }
                        >
                          <Component
                            areaChart={
                              Array [
                                Object {
                                  "color": "#DB1374",
                                  "key": "uniqueSourceIpsHistogram",
                                  "value": Array [
                                    Object {
                                      "x": 1556686800000,
                                      "y": 580213,
                                    },
                                    Object {
                                      "x": 1556730000000,
                                      "y": 1096175,
                                    },
                                    Object {
                                      "x": 1556773200000,
                                      "y": 12382,
                                    },
                                  ],
                                },
                                Object {
                                  "color": "#490092",
                                  "key": "uniqueDestinationIpsHistogram",
                                  "value": Array [
                                    Object {
                                      "x": 1556686800000,
                                      "y": 565975,
                                    },
                                    Object {
                                      "x": 1556730000000,
                                      "y": 1084366,
                                    },
                                    Object {
                                      "x": 1556773200000,
                                      "y": 12280,
                                    },
                                  ],
                                },
                              ]
                            }
                          >
                            <AutoSizer
                              content={true}
                              detectAnyWindowResize={false}
                            >
                              <styled.div
                                data-test-subj="wrapped-by-auto-sizer"
                                innerRef={[Function]}
                              >
                                <div
                                  className="sc-htpNat loNVyu"
                                  data-test-subj="wrapped-by-auto-sizer"
                                >
                                  <pure(Component)
                                    data={
                                      Array [
                                        Object {
                                          "color": "#DB1374",
                                          "key": "uniqueSourceIpsHistogram",
                                          "value": Array [
                                            Object {
                                              "x": 1556686800000,
                                              "y": 580213,
                                            },
                                            Object {
                                              "x": 1556730000000,
                                              "y": 1096175,
                                            },
                                            Object {
                                              "x": 1556773200000,
                                              "y": 12382,
                                            },
                                          ],
                                        },
                                        Object {
                                          "color": "#490092",
                                          "key": "uniqueDestinationIpsHistogram",
                                          "value": Array [
                                            Object {
                                              "x": 1556686800000,
                                              "y": 565975,
                                            },
                                            Object {
                                              "x": 1556730000000,
                                              "y": 1084366,
                                            },
                                            Object {
                                              "x": 1556773200000,
                                              "y": 12280,
                                            },
                                          ],
                                        },
                                      ]
                                    }
                                  >
                                    <Component
                                      data={
                                        Array [
                                          Object {
                                            "color": "#DB1374",
                                            "key": "uniqueSourceIpsHistogram",
                                            "value": Array [
                                              Object {
                                                "x": 1556686800000,
                                                "y": 580213,
                                              },
                                              Object {
                                                "x": 1556730000000,
                                                "y": 1096175,
                                              },
                                              Object {
                                                "x": 1556773200000,
                                                "y": 12382,
                                              },
                                            ],
                                          },
                                          Object {
                                            "color": "#490092",
                                            "key": "uniqueDestinationIpsHistogram",
                                            "value": Array [
                                              Object {
                                                "x": 1556686800000,
                                                "y": 565975,
                                              },
                                              Object {
                                                "x": 1556730000000,
                                                "y": 1084366,
                                              },
                                              Object {
                                                "x": 1556773200000,
                                                "y": 12280,
                                              },
                                            ],
                                          },
                                        ]
                                      }
                                    />
                                  </pure(Component)>
                                </div>
                              </styled.div>
                            </AutoSizer>
                          </Component>
                        </pure(Component)>
                      </div>
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                </div>
              </EuiFlexGroup>
            </div>
          </EuiPanel>
        </div>
      </EuiFlexItem>
    </Styled(EuiFlexItem)>
  </Component>
</pure(Component)>
`;

exports[`Stat Items rendering kpis without charts it renders the default widget 1`] = `
<pure(Component)
  description="HOSTS"
  fields={
    Array [
      Object {
        "color": "#3185FC",
        "key": "hosts",
        "value": null,
      },
    ]
  }
  isLoading={false}
>
  <Component
    description="HOSTS"
    fields={
      Array [
        Object {
          "color": "#3185FC",
          "key": "hosts",
          "value": null,
        },
      ]
    }
    isLoading={false}
  >
    <Styled(EuiFlexItem)
      key="stat-items-undefined"
    >
      <EuiFlexItem
        className="sc-bxivhb bQCHAr"
      >
        <div
          className="euiFlexItem sc-bxivhb bQCHAr"
        >
          <EuiPanel
            grow={true}
            hasShadow={false}
            paddingSize="m"
          >
            <div
              className="euiPanel euiPanel--paddingMedium"
            >
              <EuiTitle
                size="xxxs"
              >
                <h6
                  className="euiTitle euiTitle--xxxsmall"
                >
                  HOSTS
                </h6>
              </EuiTitle>
              <EuiFlexGroup>
                <div
                  className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <Styled(EuiFlexItem)
                    key="stat-items-field-hosts"
                  >
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      >
                        <EuiFlexGroup
                          alignItems="center"
                          gutterSize="m"
                          responsive={false}
                        >
                          <div
                            className="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                          >
                            <Styled(EuiFlexItem)>
                              <EuiFlexItem
                                className="sc-bxivhb bQCHAr"
                              >
                                <div
                                  className="euiFlexItem sc-bxivhb bQCHAr"
                                >
                                  <Styled(EuiTitle)>
                                    <EuiTitle
                                      className="sc-ifAKCX bfXlPM"
                                    >
                                      <p
                                        className="euiTitle euiTitle--medium sc-ifAKCX bfXlPM"
                                      >
                                         
                                      </p>
                                    </EuiTitle>
                                  </Styled(EuiTitle)>
                                </div>
                              </EuiFlexItem>
                            </Styled(EuiFlexItem)>
                          </div>
                        </EuiFlexGroup>
                      </div>
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                </div>
              </EuiFlexGroup>
              <EuiFlexGroup>
                <div
                  className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                />
              </EuiFlexGroup>
            </div>
          </EuiPanel>
        </div>
      </EuiFlexItem>
    </Styled(EuiFlexItem)>
  </Component>
</pure(Component)>
`;

exports[`Stat Items rendering kpis without charts it renders the default widget 2`] = `
<pure(Component)
  areaChart={Array []}
  barChart={Array []}
  description="HOSTS"
  fields={
    Array [
      Object {
        "color": "#3185FC",
        "key": "hosts",
        "value": null,
      },
    ]
  }
  isLoading={false}
>
  <Component
    areaChart={Array []}
    barChart={Array []}
    description="HOSTS"
    fields={
      Array [
        Object {
          "color": "#3185FC",
          "key": "hosts",
          "value": null,
        },
      ]
    }
    isLoading={false}
  >
    <Styled(EuiFlexItem)
      key="stat-items-undefined"
    >
      <EuiFlexItem
        className="sc-bxivhb bQCHAr"
      >
        <div
          className="euiFlexItem sc-bxivhb bQCHAr"
        >
          <EuiPanel
            grow={true}
            hasShadow={false}
            paddingSize="m"
          >
            <div
              className="euiPanel euiPanel--paddingMedium"
            >
              <EuiTitle
                size="xxxs"
              >
                <h6
                  className="euiTitle euiTitle--xxxsmall"
                >
                  HOSTS
                </h6>
              </EuiTitle>
              <EuiFlexGroup>
                <div
                  className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <Styled(EuiFlexItem)
                    key="stat-items-field-hosts"
                  >
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      >
                        <EuiFlexGroup
                          alignItems="center"
                          gutterSize="m"
                          responsive={false}
                        >
                          <div
                            className="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                          >
                            <Styled(EuiFlexItem)>
                              <EuiFlexItem
                                className="sc-bxivhb bQCHAr"
                              >
                                <div
                                  className="euiFlexItem sc-bxivhb bQCHAr"
                                >
                                  <Styled(EuiTitle)>
                                    <EuiTitle
                                      className="sc-ifAKCX bfXlPM"
                                    >
                                      <p
                                        className="euiTitle euiTitle--medium sc-ifAKCX bfXlPM"
                                      >
                                         
                                      </p>
                                    </EuiTitle>
                                  </Styled(EuiTitle)>
                                </div>
                              </EuiFlexItem>
                            </Styled(EuiFlexItem)>
                          </div>
                        </EuiFlexGroup>
                      </div>
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                </div>
              </EuiFlexGroup>
              <EuiFlexGroup>
                <div
                  className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                >
                  <Styled(EuiFlexItem)>
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      />
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                  <Styled(EuiFlexItem)>
                    <EuiFlexItem
                      className="sc-bxivhb bQCHAr"
                    >
                      <div
                        className="euiFlexItem sc-bxivhb bQCHAr"
                      />
                    </EuiFlexItem>
                  </Styled(EuiFlexItem)>
                </div>
              </EuiFlexGroup>
            </div>
          </EuiPanel>
        </div>
      </EuiFlexItem>
    </Styled(EuiFlexItem)>
  </Component>
</pure(Component)>
`;
